packages/utilities/src/lib/utilityTypes.ts

```diff
@@ -72,3 +72,91 @@ export type Awaited<T> = PromiseLike<T> | T;
  * Type union for the full 2 billion dollar mistake in the JavaScript ecosystem
  */
 export type Nullish = null | undefined;
+
+/**
+ * Similar to the built in {@link NonNullable}, but properly removes `null` from all keys in the class or interface
+ * This does not recurse deeply, for that use {@link DeepRequired}
+ */
+export type NonNullableClass<T = unknown> = {
```

Comment:
Maybe we should use NonNullableProperties?

Reply:
what now?